src/App.tsx

```diff
@@ -1,61 +1,127 @@
+import React, { useState } from 'react';
+import { findById, addId } from './helpers';
+import { User, Todos } from './types/types';
+import { TodoList } from './components/TodoList';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const todos = todosFromServer.map(todo => {
+  const user = usersFromServer.find(
+    findById<User>(todo.userId),
+  ) || null;
+
+  return {
+    ...todo,
+    user,
+  };
+});
+
+export const App: React.FC = () => {
+  const [todosList, setTodosList] = useState<Todos[]>(todos);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [isTitleEmpty, setIsTitleEmpty] = useState(false);
+  const [isIdSelected, setIsIdSelected] = useState(false);
+
+  const newTodo: Todos = {
+    id: addId(todosList),
+    title,
+    completed: false,
+    userId: selectedUserId,
+    user: usersFromServer.find(
+      findById<User>(selectedUserId),
+    ) || null,
+  };
+
+  const handleValidation = () => {
+    setIsTitleEmpty(!title);
+    setIsIdSelected(!selectedUserId);
+  };
+
+  const clearForm = () => {
+    setTitle('');
+    setSelectedUserId(0);
+  };
+
+  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    handleValidation();
+
+    if (!title || !selectedUserId) {
+      return;
+    }
+
+    setTodosList(prevTodosList => [...prevTodosList, newTodo]);
+    clearForm();
+  };
+
+  const handleInputTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
```

Comment:
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364
```suggestion
  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
```